sycl/include/sycl/ext/intel/experimental/esimd/detail/memory_intrin.hpp

```diff
@@ -687,48 +719,109 @@ template <typename Ty, int M, int N, typename TACC>
 inline __SEIEED::vector_type_t<Ty, M * N>
 __esimd_media_block_load(unsigned modififer, TACC handle, unsigned plane,
                          unsigned width, unsigned x, unsigned y) {
-  // On host the input surface is modeled as sycl image 2d object,
-  // and the read/write access is done through accessor,
-  // which is passed in as the handle argument.
-  auto range = __SEIEED::AccessorPrivateProxy::getImageRange(handle);
-  unsigned bpp = __SEIEED::AccessorPrivateProxy::getElemSize(handle);
-  unsigned vpp = bpp / sizeof(Ty);
-  unsigned int i = x / bpp;
-  unsigned int j = y;
-
-  assert(x % bpp == 0);
-  unsigned int xbound = range[0] - 1;
-  unsigned int ybound = range[1] - 1;
-
   __SEIEED::vector_type_t<Ty, M * N> vals;
-  for (int row = 0; row < M; row++) {
-    for (int col = 0; col < N; col += vpp) {
-      unsigned int xoff = (i > xbound) ? xbound : i;
-      unsigned int yoff = (j > ybound) ? ybound : j;
-      auto coords = cl::sycl::cl_int2(xoff, yoff);
-      cl::sycl::cl_uint4 data = handle.read(coords);
-
-      __SEIEED::vector_type_t<unsigned int, 4> res;
-      for (int idx = 0; idx < 4; idx++) {
-        res[idx] = data[idx];
+
+  sycl::detail::ESIMDDeviceInterface *I =
+      sycl::detail::getESIMDDeviceInterface();
```

Comment:
This change causes failures from following test with 'host' device
- SYCL/ESIMD/histogram.cpp
- SYCL/ESIMD/histogram_2d.cpp
- SYCL/ESIMD/histogram_raw_send.cpp
- SYCL/ESIMD/linear/linear.cpp
- SYCL/ESIMD/vadd_2d.cpp

This is because with 'host' device loaded, `getPluginOpaqueData<backend::esimd_cpu>()` inside of `getESIMDDeviceInterface` should fail as `getPluginOpaqueData` calls `getPlugin<backend::esimd_cpu>()` (Failure code : 'pi::getPlugin couldn't find plugin -59').

Unless there is a way to distinguish between 'host' type and 'esimd_cpu' type at runtime within this intrinsic implementation, I think these failing samples must be marked as UNSUPPORTED for host device type.

Reply:
@kbobrovs , Can I mark host-device unsupported for above ESIMD tests in intel/llvm-test-suite?

Reply:
PR created for marking UNSUPPORTED for failing ones : https://github.com/intel/llvm-test-suite/pull/492

Reply:
I wonder how other tests pass then, like vadd_1d? Isn't the PI call sequence the same for them?

Reply:
Other passing tests like `vadd_1d` either do not have `HOST_RUN_PLACEHOLDER` line. (https://github.com/intel/llvm-test-suite/blob/1a770de1b40f9cce67b63d0e42ac5d06984f7ce3/SYCL/ESIMD/vadd_1d.cpp#L11)

My changes in this file are applied only for `#ifndef SYCL_DEVICE_ONLY` therefore only host mode execution is impacted - irrelevant to PI call sequence. (ESIMD_CPU build is disabled by default in origin/sycl)

There are 17 samples with the `HOST_RUN_PLACEHOLDER` line from `SYCL/ESIMD/` directory.
```SYCL/ESIMD/BitonicSortKv2.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/PrefixSum.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out 20
SYCL/ESIMD/Prefix_Local_sum1.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out 20
SYCL/ESIMD/Prefix_Local_sum2.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out 20
SYCL/ESIMD/Prefix_Local_sum3.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/Stencil.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/histogram.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/histogram_2d.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/histogram_raw_send.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/kmeans/kmeans.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out %S/points.csv
SYCL/ESIMD/linear/linear.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out %S/linear_in.bmp %S/linear_gold_hw.bmp
SYCL/ESIMD/matrix_transpose_usm.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/reduction.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/stencil2.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/test_id_3d.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/vadd_2d.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
SYCL/ESIMD/vadd_usm.cpp:// RUN: %HOST_RUN_PLACEHOLDER %t.out
```
Among these 17 samples, the above five samples are impacted by this change set - all of them due to `media_block_load/store`